### PR TITLE
Add tabs proposal and research skeleton docs

### PR DESCRIPTION
**Pull request by matthewp**
These are just skeletons and not filled out. Intend to iterate on these a little at a time.

### CODE REVIEW

File: research/src/pages/tabs.proposal.mdx
Comment (gregwhitworth) [inline research/src/pages/tabs.proposal.mdx:1]: Can you add the header for this:

```
---
menu: Proposals
name: Tabs (Editor's Draft)
path: /components/tabs
pathToResearch: /components/tabs.research
showInMenu: false
---
```

File: research/src/pages/tabs.research.mdx
Comment (gregwhitworth) [inline research/src/pages/tabs.research.mdx:2]: I'm making a change to the nav structure to handle this type of scenario. Also, the research doesn't progress like a spec so it doesn't have a stage.

```
---
name: Tabs
menu: Analysis
path: /components/tabs.research
pathToProposal: /components/tabs
---
```

File: research/src/pages/tabs.research.mdx
Comment (gregwhitworth) [inline research/src/pages/tabs.research.mdx:13]: I should just remove this from the template as this ends up in the proposal and is an artifact that is left over from early on

**Review by gregwhitworth — CHANGES_REQUESTED** [review]

**Comment by matthewp** [comment]
Updated with the requested changes.

File: research/src/pages/tabs.proposal.mdx
Comment (bkardell) [inline research/src/pages/tabs.proposal.mdx:13]: Is it worth mentioning that they are well defined in accessibility trees and long defined as roles and relationships in ARIA?

File: research/src/pages/tabs.proposal.mdx
Comment (bkardell) [inline research/src/pages/tabs.proposal.mdx:23]: I'm not suggesting we change this before merging but I feel like we can add/find better use case docs in OS guides maybe?  This sounds awfully like someone will say "that's a wizard" not tabs or something

File: research/src/pages/tabs.proposal.mdx
Comment (bkardell) [inline research/src/pages/tabs.proposal.mdx:32]: Strange question, but one I got asked in the past so should probably carry over: Are there any kinds of limits or definition of supplemental content?  Can it be other interactive elements?  A close or dismiss button is the most obvious example here.

File: research/src/pages/tabs.proposal.mdx
Comment (bkardell) [inline research/src/pages/tabs.proposal.mdx:35]: In discussion with some developers recently it was suggested that this was less than pleasing in practice actually, and this matches my own experience from implementing and using this as elements many times.  Things that do this by ref attribute are more pleasing is the alternative feedback we've gotten - in other words, mark the tab's related 'element' as active with a boolean attribute or something or a method that takes the tab to make active. /me shrugs

**Comment by bkardell** [comment]
I made some comments here mostly as notes to refer back to but this lgtm as a starting point for discussion and I think we should merge it.

File: research/src/pages/tabs.proposal.mdx
Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:35]: Sounds like this deserves an issue all of its own.

File: research/src/pages/tabs.proposal.mdx
Comment (chrisdholt) [reply research/src/pages/tabs.proposal.mdx:32]: We dug quite a bit to try and find a pre-existing answer to this exact question. We have a couple of specific scenarios in FAST we needed to be able to support where there are static interactive elements to the right or left of the tab list itself. The intent with the note above is to specifically call out that this would be supported either to the right or left of the tablist - not part of it. In the case something interferes with the tabs interaction model, I think interactive content would be problematic. We decided that so long as you can still successfully navigate the tabs model per the W3C ARIA specification, there's no problem including interactive elements in those slots.

File: research/src/pages/tabs.proposal.mdx
Comment (bkardell) [reply research/src/pages/tabs.proposal.mdx:35]: most of them probably do, as I said they are mostly notes for me to remember to come back once merged

**Comment by bkardell** [comment]
@gregwhitworth did you see @matthewp's changes, it looks like it resolved your review but it seems it says it is blocked on requested changes? I think that's you?

**Comment by gregwhitworth** [comment]
@bkardell I try to get in at least 1-2 times a week to triage and review items. Ironically Wed is one of those pre-defined days. With regards to the review itself there is a discussion between @chrisdholt and @matthewp above. So I'm not going to sign off on that until a new issue is opened to track the potential resolutions. So actions here:

1. @matthewp / @bkardell can you please open issues that are not blocking this PR so they can be discussed independant of this
2. @chrisdholt can you submit a review of this PR and if your feedback isn't blocking file an issue regarding it and resolve the discussion. If it is blocking please mark it accordingly in the review

Thanks

**Comment by bkardell** [comment]
@gregwhitworth can you point to somewhere where this is guidance on how you'd like to do this stuff procedurally? My understanding was that this is a vague starting point on which there should be plenty of discussions, further work and issues opened against from lots of interested parties.  I made notes here specifically so I could come back and open issues once it is merged and I have a thing to them point at.  Pointing at file that only exists in a PR seems weird to me - but if that's what you want me to do I guess I can?

**Review by chrisdholt — APPROVED** [review]

**Comment by chrisdholt** [comment]
@gregwhitworth sure thing - my feedback was primarily in response to a couple specific questions that came up. I think one is tentatively resolved unless there is push back and an issue is required (slots for interactive content at start/end of tab list). Aside from that, nothing blocking on my end here.

File: research/src/pages/tabs.proposal.mdx
Comment (una) [reply research/src/pages/tabs.proposal.mdx:13]: I think it's always worth mentioning, but perhaps there should be an ARIA/a11y section later on?

File: research/src/pages/tabs.proposal.mdx
Comment (chrisdholt) [inline research/src/pages/tabs.proposal.mdx:219]: Per @una - We should point / link to the W3C specification keyboard interaction here for sure. Perhaps another callout would help, but I think we'll want to at least point to that existing model here.

File: research/src/pages/tabs.proposal.mdx
Comment (una) [inline research/src/pages/tabs.proposal.mdx:13]: Tabs are a common mobile pattern that essentially define "sections". Many of the [Material](https://material.io/components/tabs#anatomy) examples show them this way. Therefore I would hesitate to state they are always "layered", as in some mobile UI's (a 2-tab interface), they could slide

Comment (chrisdholt) [reply research/src/pages/tabs.proposal.mdx:13]: We should at least reference / link to the interaction model below in the a11y section. Added a note there as well.

File: research/src/pages/tabs.proposal.mdx
Comment (una) [reply research/src/pages/tabs.proposal.mdx:23]: Another idea for an example are mobile patterns to indicate sections: I.e. the Twitter app Explore page

File: research/src/pages/tabs.proposal.mdx
Comment (chrisdholt) [reply research/src/pages/tabs.proposal.mdx:13]: This is good feedback; there is likely a better word than "layered" which is less design-oriented.

File: research/src/pages/tabs.proposal.mdx
Comment (una) [inline research/src/pages/tabs.proposal.mdx:47]: I would replace this with the official Material Design docs (https://material.io/components/tabs#anatomy)

File: research/src/pages/tabs.proposal.mdx
Comment (una) [inline research/src/pages/tabs.proposal.mdx:207]: Also consider touch events such as swipe

File: research/src/pages/tabs.proposal.mdx
Comment (una) [inline research/src/pages/tabs.proposal.mdx:224]: Additional questions:

- What is the behavior around disabled tabs?
- What is the behavior around tabs that move around in the UI (i.e. using flexbox order to move the active tab to be "first" in the tabgroup?

File: research/src/pages/tabs.proposal.mdx
Comment (una) [inline research/src/pages/tabs.proposal.mdx:242]: I would add Tab Labels would also flip (i.e. icons next to a label)

Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:242]: Can you provide some text you'd like to see here? I'm not exactly sure what you mean but happy to add it.

File: research/src/pages/tabs.proposal.mdx
Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:224]: I'm going to open these questions as issues.

Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:224]: Actually I think I can inline these questions into this section as well and we can reference them in an issue.

File: research/src/pages/tabs.proposal.mdx
Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:47]: I think it might be good to have both. Eventually this section will have many popular tabs implementations that we can reference to talk about how these solve the various challenges. So maybe splitting this into Design Systems and Tabs Implementations makes sense?

File: research/src/pages/tabs.proposal.mdx
Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:13]: I'm going to just remove the word layered for now.

Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:13]: I added this, thanks.

File: research/src/pages/tabs.proposal.mdx
Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx]: Added, thanks!

File: research/src/pages/tabs.proposal.mdx
Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:234]: Added these questions, thanks!

File: research/src/pages/tabs.proposal.mdx
Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:225]: Did this, thanks!

File: research/src/pages/tabs.proposal.mdx
Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:23]: https://github.com/WICG/open-ui/issues/189

File: research/src/pages/tabs.proposal.mdx
Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:32]: https://github.com/WICG/open-ui/issues/190

File: research/src/pages/tabs.proposal.mdx
Comment (matthewp) [reply research/src/pages/tabs.proposal.mdx:35]: https://github.com/WICG/open-ui/issues/191

**Comment by matthewp** [comment]
@gregwhitworth I've updated the content based on feedback from @una, @bkardell and @chrisdholt. Additionally I've opened the below issues based on questions to be answered later:

* https://github.com/WICG/open-ui/issues/189
* https://github.com/WICG/open-ui/issues/190
* https://github.com/WICG/open-ui/issues/191

That leaves this one review item that I'm not sure how to resolve yet: https://github.com/WICG/open-ui/pull/183#pullrequestreview-504243913

Otherwise __(1)__ and __(2)__ are complete.

File: research/src/pages/tabs.proposal.mdx
Comment (gregwhitworth) [reply research/src/pages/tabs.proposal.mdx:252]: Opened an issue here as I don't think this needs to block the PR since this is an editor's draft initial proposal that isn't even linked to on the main page

Comment (gregwhitworth) [reply research/src/pages/tabs.proposal.mdx:252]: https://github.com/WICG/open-ui/issues/194

**Pull request merged**
